src/pilot/assign_role.py

```diff
@@ -406,8 +431,12 @@ def define_storage_logical_disks(drac_client, raid_controller_name):
             "configuration")
         return None
 
-    os_logical_disk = define_storage_operating_system_logical_disk(
-        raid_cntlr_physical_disks, drac_client, raid_controller_name)
+    if boss_controller:
+        os_logical_disk = define_storage_operating_system_logical_disk(
+            boss_cntrl_disks, drac_client, boss_controller)
+    else:
+        os_logical_disk = define_storage_operating_system_logical_disk(
+            raid_cntlr_physical_disks, drac_client, raid_controller)
```

Comment:
Here, we could just pick the first RAID controller that has 2 or more disks and pass it in including the physical disks that it controls.